BUILD.bazel

```diff
@@ -3,8 +3,15 @@ load("@rules_cc//cc:defs.bzl", "cc_library")
 
 # Header-only rule to export catch2/catch.hpp.
 cc_library(
-  name = "catch2",
-  hdrs = ["single_include/catch2/catch.hpp"],
-  visibility = ["//visibility:public"],
-  includes = ["single_include/"],
+    name = "catch2",
+    hdrs = ["single_include/catch2/catch.hpp"],
+    includes = ["single_include/"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "catch2_main",
```

Comment:
Is the expectation here that user links against both `catch2` and `catch2_main`, or will linking against `catch2_main` also bring in `includes` from `catch2`, since it is a dependency?

If the second, I would prefer if the name was `catch2_with_main` (I assume bazel uses snake_case), the way v3 does.

Reply:
only catch2_main would be needed, so went with your proposal 👍